app/core/EngineService/EngineService.ts

```diff
@@ -133,20 +133,16 @@ export class EngineService {
       return;
     }
 
-    engine.controllerMessenger.subscribeOnceIf(
-      'ComposableController:stateChange',
-      () => {
-        if (!engine.context.KeyringController.metadata.vault) {
-          Logger.log('keyringController vault missing for INIT_BG_STATE_KEY');
-        }
-        this.updateBatcher.add(INIT_BG_STATE_KEY);
-        // immediately flush the redux action
-        // so that the initial state is available to the redux store
-        this.updateBatcher.flush();
-        this.engineInitialized = true;
-      },
-      () => !this.engineInitialized,
-    );
+    if (!this.engineInitialized) {
+      if (!engine.context.KeyringController.metadata.vault) {
```

Comment:
Should this be `state`?
```suggestion
      if (!engine.context.KeyringController.state.vault) {
```

Reply:
Hmm. Yeah maybe so, though it was `metadata` before as well. I don't understand why we'd want to check the metadata here though.